src/utils.js

```diff
@@ -6,3 +6,30 @@ export function getEmojiFlag( countryCode ) {
 			.map( ( char ) => 127397 + char.charCodeAt() )
 	);
 }
+
+/*
+	I wasn't sure what XWP would name this utility function.
+	filterExcerpt? getExcerpt(seems incorrect)
+	So I went with formatExcerpt
+
+	Function comments
+	Accepts:
+		excerpt
+		excerptLength
+	Returns:
+		If excerpt is longer than excerptLength
+			Excerpt without HTML trimmed to desired length ending in ...
+		Else 
+			Empty string
+*/
+export function formatExcerpt( excerpt, excerptLength ) {
+	if ( ! excerpt ) return '';
+
+	if ( excerpt.length > excerptLength ) {
+		return (
+			excerpt.replace( /<[^>]*>?/gm, '' ).substring( 0, excerptLength ) +
+			'...'
+		);
+	}
+	return '';
+}
```

Comment:
@cdevroe What are the other ways to sanitize HTML markup in JavaScript? Assume that the client wants to preserve HTML tags in the output.

Reply:
If there are built in WordPress JavaScript utility functions to do something like this I couldn't find one. One way could be to create a list of accepted HTML tags that would be allowed in the Excerpt. Such as EM, STRONG, etc. And after I’ve truncated the Excerpt to the appropriate size, close any unclosed HTML tags found in the truncated excerpt. There is a simple, yet perhaps hacky?, way to do this by creating a function that uses `.innerHTML` to do so.